core/src/main/java/org/apache/iceberg/GenericPartitionFieldSummary.java

```diff
@@ -135,9 +136,9 @@ public Object get(int i) {
       case 0:
         return containsNull;
       case 1:
-        return lowerBound;
+        return lowerBound();
```

Comment:
I return `ByteBuffer` by calling `lowerBound()` even though the internal representation is `byte[]`. I think it is different from `GenericDataFile`. Did we overlook `get` for `keyMetadata` in `GenericDataFile`, @rdblue?

Reply:
Yes, I think we overlooked that option. That sounds cleaner if you want to make the change.

Reply:
I've created #728 to address that.